### ISSUE TITLE
/updates returns 500

### ISSUE THREAD

**Issue by jharting** [RedHatInsights/vmaas#203]
If I run 
```
curl -v http://webapp-vmaas-stable.1b13.insights.openshiftapps.com/api/v1/updates -X POST -d '{"package_list": ["hwdata-0.252-8.4.el7.x86_64"]}'
```

I am getting 500: Internal Server Error.

Similar requests such as 
```
curl -v http://webapp-vmaas-stable.1b13.insights.openshiftapps.com/api/v1/updates -X POST -d '{"package_list": ["cpp-4.8.5-11.el7.x86_64"]}'
```
seem to work fine.

**Comment by jharting** [comment]
These other packages seem to cause the API to fail:
```
"rsyslog-7.4.7-16.el7.x86_64"
"iwl2000-firmware-18.168.6.1-49.el7.noarch"
"linux-firmware-20160830-49.git7534e19.el7.noarch"
"iwl2030-firmware-18.168.6.1-49.el7.noarch"
"iwl4965-firmware-228.61.2.24-49.el7.noarch"                                                                                                                                                                                                
"iwl3160-firmware-22.0.7.0-49.el7.noarch"                                                                                                                                                                                                   
"iwl6050-firmware-41.28.5.1-49.el7.noarch"                                                                                                                                                                                                  
"iwl135-firmware-18.168.6.1-49.el7.noarch"                                                                                                                                                                                                  
"iwl6000g2a-firmware-17.168.5.3-49.el7.noarch"                                                                                                                                                                                              
"iwl3945-firmware-15.32.2.9-49.el7.noarch"                                                                                                                                                                                                  
"iwl105-firmware-18.168.6.1-49.el7.noarch"                                                                                                                                                                                                  
"iwl5000-firmware-8.83.5.1_1-49.el7.noarch"                                                                                                                                                                                                 
"iwl100-firmware-39.31.5.1-49.el7.noarch"                                                                                                                                                                                                   
"iwl6000g2b-firmware-17.168.5.2-49.el7.noarch"                                                                                                                                                                                              
"iwl7260-firmware-22.0.7.0-49.el7.noarch"                                                                                                                                                                                                   
"iwl5150-firmware-8.24.2.2-49.el7.noarch"                                                                                                                                                                                                   
"iwl6000-firmware-9.221.4.1-49.el7.noarch"                                                                                                                                                                                                  
"hwdata-0.252-8.4.el7.x86_64"
```

**Comment by jdobes** [comment]
In logs:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/tornado/web.py", line 1412, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/app/app.py", line 199, in post
    self.process_post()
  File "/app/app.py", line 108, in process_post
    response = self.process_list(data)
  File "/app/app.py", line 130, in process_list
    return self.application.updatesapi.process_list(data)
  File "/app/updates.py", line 292, in process_list
    for r_id in pkg_id2repo_id[upd_pkg_id]:
KeyError: 4708
```

This is caused by synced Fedora repos - they got updated - packages are now missing in remote repo but are only disassociated from repo in our DB (until #22 is fixed). Then it hits bug in code during filtering possible updates. Let's fix this and backport to v0.3.
